addons/stock/models/stock_inventory.py

```diff
@@ -99,8 +99,8 @@ def _compute_total_qty(self):
     @api.multi
     def unlink(self):
         for inventory in self:
-            if inventory.state == 'done':
-                raise UserError(_('You cannot delete a validated inventory adjustement.'))
+            if inventory.state not in ('draft', 'cancel'):
+                raise UserError(_('You cannot delete an inventory adjustement which is not draft or cancelled.'))
```

Comment:
that breaks the existing translations, see https://github.com/odoo/odoo/wiki/Contributing#how-should-i-handle-translatable-content